third_party/tonic/dart_state.h

```diff
@@ -68,6 +68,9 @@ class DartState : public std::enable_shared_from_this<DartState> {
   void SetReturnCodeCallback(std::function<void(uint32_t)> callback);
   bool has_set_return_code() const { return has_set_return_code_; }
 
+  void SetIsShuttingDown() { is_shutting_down_ = true; }
```

Comment:
I'm not familiar with the threading of this code, would you want an atomic bool here?

Reply:
Done